main.sh

```diff
@@ -54,13 +54,19 @@ azure_login() {
   fi
 
   # Log in to Azure using a service principal
-  az login --service-principal -u "${ARM_CLIENT_ID}" -p "${ARM_CLIENT_SECRET}" --tenant "${ARM_TENANT_ID}" &>/dev/null
-  if [[ $? -ne 0 ]]; then
-    err "Failed to authenticate with Azure. Please check your Cloud Connector configuration."
+  if ! login_output=$(az login --service-principal -u "${ARM_CLIENT_ID}" -p "${ARM_CLIENT_SECRET}" --tenant "${ARM_TENANT_ID}" 2>&1); then
```

Comment:
this will never is false I think as you are assigning a value to `login_output` not checking the output of the command. Also no need to store the login_output, just throw the error to the std err and when redirect the std out to dev null. Too complex right now, I think.

Same for below

Reply:
got my answers. I was wrong.